framework/widgets/ActiveField.php

```diff
@@ -775,19 +775,19 @@ public function widget($class, $config = [])
         $config['attribute'] = $this->attribute;
         $config['view'] = $this->form->getView();
         if (is_subclass_of($class, 'yii\widgets\InputWidget')) {
-            foreach ($this->inputOptions as $key => $value) {
-                if (!isset($config['options'][$key])) {
-                    $config['options'][$key] = $value;
-                }
-            }
+            $classOptions = (new \ReflectionClass($class))->getDefaultProperties();
```

Comment:
Reflecting every widget class isn't particularly good performance-wise. Is there another way?

Reply:
Yes, that's right. I didn't find a one. Do you have any ideas?

Reply:
No, not at the moment.

Reply:
@samdark @samdark `get_class_vars($class)` and native `foreach ($this as $prop => $value)` gives same result. but i think u miss `behaviors()`

Reply:
> get_class_vars($class) and native foreach ($object as $prop => $value) gives same result.

That's reflection as well.

> but i think u miss behaviors()

What do you mean?

Reply:
@samdark behaviors appends properties and methods